Update gitlab token setting and enhance .gitignore

KonstantinVasilkov opened the pull request:
Changed the way to receive the GITLAB API token within the `stb/config.py`. Token information is now retrieved via the Typer prompt. Ensures token data is present before setting up local services, thus reducing errors, and improving usability. Further, ignored all Pycharm-related files & directories (`.idea/`) making the code base cleaner for Pycharm users.

Commit:
Update gitlab token setting and enhance .gitignore

Changed the way to receive the GITLAB API token within the `stb/config.py`. Token information is now retrieved via the Typer prompt. Ensures token data is present before setting up local services, thus reducing errors, and improving usability. Further, ignored all Pycharm-related files & directories (`.idea/`) making the code base cleaner for Pycharm users.

zmievsa commented on stb/config.py:
```diff
@@ -165,10 +165,14 @@ def get_gitlab_api_token() -> None:
     """Set the gitlab api token name for setting up local services"""
+    if token_name is None or token is None:
+        typer.echo("Token name or token is missing. Please, run the command again.")
+        raise typer.Exit()
```
Let's exit with status code 1. It's a good practice.

KonstantinVasilkov replied:
I have improved this part to make it more interactive. Now we are providing a user instruction first where he can get the API token, and only after that, we are offering him to enter values. Check values are not None useless, as the typer is not allowing a user to input a blank string, but I have added a check that user input does not contain only whitespaces.

zmievsa reviewed (dismissed):
Big thanks for your PR! I left just one minor comment but it's quite ready to be merged.

Please, bump the major version in pyproject.toml as this is a  breaking change.

zmievsa committed:
Merge branch 'main' into main

Commit:
Refine token input prompts for GitLab API

Improved the user experience in the setup process for the GitLab API by adding a validation function ('no_whitespace') to the input prompts for user token and token name. The function checks if the inputs are either empty or only contains whitespaces, triggering an error if so. This prevents empty or whitespace-only entries from being accepted, which were previously causing errors in the setup process. Also welcome prompt was added to notify user how to obtain gitlab API token.

Commit:
Merge remote-tracking branch 'origin/main'
